vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/test/grid-connector-data-range.test.ts

```diff
@@ -5,27 +5,44 @@ import {
 } from './shared.js';
 import type { FlowGrid } from './shared.js';
 
-const SIZE = 200;
 const PAGE_SIZE = 50;
 
 describe('grid connector - data range', () => {
   let grid: FlowGrid;
-  let lastRange: [number, number] | null = null;
+  let lastRequestedRange: [number, number] | null;
+  let rootSize: number;
 
-  function setItemsRange(start: number, count: number) {
-    const items = Array.from({ length: SIZE }, (_, i) => ({ key: `${i}`, name: `Item ${i}` }));
+  function setRootItemsRange(start: number, count: number) {
+    const items = Array.from({ length: rootSize }, (_, i) => ({ key: `${i}`, name: `Item-${i}` }));
 
-    if (lastRange) {
-      grid.$connector.clear(lastRange[0], lastRange[1]);
+    grid.$connector.updateSize(rootSize);
+
+    if (lastRequestedRange) {
+      grid.$connector.clear(lastRequestedRange[0], lastRequestedRange[1]);
     }
 
+    count = Math.min(count, rootSize - start);
```

Comment:
**note:** I had to add this workaround because the connector currently adds 1 extra page to the visible range when you scroll to the end. Flow then discards that page from the range on the server-side.